frontend/js/utils/getFormData.js

```diff
@@ -83,7 +83,7 @@ export const gatherRepeaters = (rootState) => {
         delete repeaterBlock.type
 
         // and lastly we want to keep the id to update existing items
-        repeaterBlock.id = repeaterItem.id
+        fields.id = repeaterItem.id
```

Comment:
@yanhao-li what about removing this line completely?  `repeaterBlock` should have been assigned `repeaterItem.id` in `id` by `buildBlock` already, but I might be missing something! Can you try it out? Thanks!

Reply:
@ifox You're right, the `repeaterBlock` indeed has already been assigned `repeaterItem.id` by `buildBlock`, the tricky part here is, on line 88
```php
        return Object.assign(repeaterBlock, fields)
```

Where `fields` is from:
```php
        let fields = repeaterBlock.content
```

However, for repeaters used on static form, the `fields` will include the related entity's id from database, and it override the `id` of `repeaterBlock`.

I'm not sure if there is any specific reason that you decided to keep the `id` while building block, if not, we could choose to get rid of the `id` from `fields` instead.

The only potential problem is, if the repeater state hasn't been persisted, it will still delete old block and create a same new one, because this is not a idempotent operation, we can solve it in the future by providing a more detailed http response.

but for now, this is the easiest way I can see to solve this specific issue. I have tested the updated repeater both on static form or inside block editor, didn't find any problems.

Reply:
You're totally right, thank you so much for the investigation. I'm perfectly fine with merging this fix like this. We can improve in the future when refactoring, but we really need to get this fix in for now!